anndata/core/sparsedataset.py

```diff
@@ -8,79 +8,117 @@
 
 # TODO:
 # - think about supporting the COO format
-from collections.abc import Iterable
+import collections.abc as cabc
 from itertools import accumulate, chain
-from typing import Union, NamedTuple, Tuple, Sequence, Type
+from typing import Union, NamedTuple, Tuple, Sequence, Iterable, Type
 from warnings import warn
 
 import h5py
 import numpy as np
 import scipy.sparse as ss
 from scipy.sparse import _sparsetools
 
-from ...utils import unpack_index
+try:
+    # Not really important, just for IDEs to be more helpful
+    from scipy.sparse.compressed import _cs_matrix
+except ImportError:
+    _cs_matrix = ss.spmatrix
+
+from .index import unpack_index, Index
 
 
 class BackedFormat(NamedTuple):
     format_str: str
-    backed_type: Type['BackedSparseMatrixMixin']
+    backed_type: Type['BackedSparseMatrix']
     memory_type: Type[ss.spmatrix]
 
 
-class BackedSparseMatrixMixin:
+class BackedSparseMatrix(_cs_matrix):
```

Comment:
Your concerns have always been addressed 11-15 lines above.

Reply:
I still think we should not explicitly inherit from a private class. Someone might start using a method from this class and we wouldn't know until it broke. Also, having been looking at the code for the class we'd be inheriting from, I want to distance ourself from it.

Are there other solutions to the problem being solved? Also, what exactly is the problem being solved?

* Can this be done in a stub file?
* Could this be an abstract base class with class attributes which help?

Reply:
That’s the base class of `csr_matrix` and `csc_matrix`, we’re inheriting from that anyway farther down.

Reply:
Yeah, but there can be strange metaclasses in numpy code, and I don't think anything in the implementation of `BackedSparseMatrix` should even have the chance of relying on methods from `_cs_matrix`. If it inherits from that class, someone might try and use a method, which means that this would break with any scipy release that changes code (or renames a private class, or moves where it's defined) which has no API promises. I think it's better to head that off, since it was pretty painful for me to deal with before.

I know it's kind of doing that already by patching private methods, but I'd like to limit the reliance as much as possible.

Reply:
OK!